metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -271,9 +271,10 @@ def step_init(self, flow, graph, step, decos, environment, flow_datastore, logge
 
         for deco in decos:
             if isinstance(deco, ResourcesDecorator):
+                self.attributes["gpu"] = None
                 for k, v in deco.attributes.items():
-                    # If GPU count is specified, explicitly set it in self.attributes.
-                    if k == "gpu" and v != None:
+                    # If GPU count is specified and non-zero, explicitly set it in self.attributes.
+                    if k == "gpu" and v:
```

Comment:
looking into alternatives for this at the moment. Note that the default from resources deco is actually `"0"` and not int, so it will pass this check.

Reply:
Hmm, I wonder if it's cast before it ends up here, because the [type in the comments in `int`](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/resources_decorator.py#L26). Also, this seems to fix the problem, which is my biggest priority for removing the footgun; setting the default to `None` would also expedite the fix, so that works, as well.

I can add some logging to capture the default values, here, to demonstrate.